pkg/validation/internal/removed_apis.go

```diff
@@ -335,25 +336,56 @@ func getRemovedAPIsOn1_25From(bundle *manifests.Bundle) (map[string][]string, ma
 		}
 	}
 
-	warnIfDeprecated := func(res string, msg string) {
-		switch res {
-		case "cronjobs":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "endpointslices":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "events":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "horizontalpodautoscalers":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "poddisruptionbudgets":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "podsecuritypolicies":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
-		case "runtimeclasses":
-			warnDeprecatedAPIs[res] = append(warnDeprecatedAPIs[res], msg)
+	warnIfDeprecated := func(gr schema.GroupResource, msg string) {
```

Comment:
I wonder if there's a way to have a `map[schema.GroupResource]struct{}` that is essentially a set of the group/resources that have deprecations in their ranks.

It seems like that could cleanup a lot of the duplication I'm seeing here.

Similar comment for the GVKs in `addIfDeprecated`.

Reply:
This should be resolved as of: https://github.com/operator-framework/api/pull/274/commits/f31e203a46d8e8b7bd517952fe927865365ad2d9